internal/resolution/variablesources/bundle.go

```diff
@@ -117,3 +93,53 @@ func (b *BundlesAndDepsVariableSource) filterBundleDependencies(allBundles []*ca
 
 	return dependencies, nil
 }
+
+type BundlesAndDepsVariableSource struct {
+	catalogClient   BundleProvider
+	variableSources []input.VariableSource
+}
+
+func NewBundlesAndDepsVariableSource(catalogClient BundleProvider, inputVariableSources ...input.VariableSource) *BundlesAndDepsVariableSource {
+	return &BundlesAndDepsVariableSource{
+		catalogClient:   catalogClient,
+		variableSources: inputVariableSources,
+	}
+}
+
+func (b *BundlesAndDepsVariableSource) GetVariables(ctx context.Context) ([]deppy.Variable, error) {
+	variables := []deppy.Variable{}
+
+	for _, variableSource := range b.variableSources {
+		inputVariables, err := variableSource.GetVariables(ctx)
+		if err != nil {
+			return nil, err
+		}
+		variables = append(variables, inputVariables...)
+	}
+
+	allBundles, err := b.catalogClient.Bundles(ctx)
+	if err != nil {
+		return nil, err
+	}
```

Comment:
Help me remember. Is this call eventually being removed by the time we get to the end of the PR chain? Just want to make sure we aren't ending up with clients sourcing input under the covers anywhere.

Reply:
> Help me remember. Is this call eventually being removed by the time we get to the end of the PR chain?

Correct, I'm cheating here. Passing all bundles trough current structure is going to be challenging.

This whole variable source will go away in the final part #501: https://github.com/operator-framework/operator-controller/pull/501/commits/27dd463595641d2068d8733d280d0af4f1b8f663#diff-26d40ef0af5660e5b7fd99da5952b39c645bef2ad6bf2ee6d8b7123bbdd2ff73

Reply:
@joelanford I came up with a relatively small change so we can get rid of this: https://github.com/operator-framework/operator-controller/pull/508

This PR also enables me to progress with the last bit of implementation for initial upgrade support RFC (#449) without having to wait for part 2-5 of this refactorign to land.